Added 'extractVideoEmbedFromStory()' helper function & unit tests

vinnie-olsen opened the pull request:
## Description
Added `extractVideoEmbedFromStory()` helper function

## Jira Ticket
- [PEN-](https://arcpublishing.atlassian.net/browse/PEN-1493)

## Acceptance Criteria
_copy from ticket_

## Test Steps
Test together with steps in https://github.com/WPMedia/fusion-news-theme-blocks/pull/584

## Effect Of Changes
### Before
_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After
_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps above are working
- [X] Confirmed there are no linter errors
- [X] Confirmed this PR has reasonable code coverage
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [X] Confirmed relevant documentation has been updated/added.

vinnie-olsen committed:
Added 'extractVideoEmbedFromStory()' helper function & unit tests

vinnie-olsen committed:
Fix linting error

jgrosspietsch commented on src/utils/extractVideoEmbedFromStory.ts:
```diff
@@ -0,0 +1,16 @@
+const extractVideoEmbedFromStory = (content) => {
```
Need a return type on this. https://www.typescriptlang.org/docs/handbook/functions.html#typing-the-function
```suggestion
const extractVideoEmbedFromStory = (content): string => {
```

jgrosspietsch commented on src/utils/extractVideoEmbedFromStory.ts:
```diff
@@ -0,0 +1,16 @@
+        ? content.promo_items.lead_art : undefined
+    )
+  );
+  return contentEl ? contentEl.embed_html : undefined;
```
Had to rewrite this to understand what was going on.

```ts
const extractVideoEmbedFromStory = (content): string => {
  if (content && content.type === 'video') {
    return content.embed_html;
  }

  if (
    (content && content.type === 'story')
    && (
      content.promo_items
      && content.promo_items.lead_art
      && content.promo_items.lead_art.type === 'video'
    )
  ) {
    return content.promo_items.lead_art.embed_html;
  }

  return undefined;
};
```

Does this track with what you were going for? Tests pass for this version so I assume so. Damn shame optional chaining isn't working with our Jest setup.

vinnie-olsen replied:
Yeah I agree - I actually had this written with optional chaining before and then rewrote it without when I realized it isn't supported in this repo. I think the way you have it re-written replicates the logic but I will test it out locally to make sure everything still works as expected on the front-end 👍

vinnie-olsen committed:
Refactored 'extractVideoEmbedFromStory', added interface

vinnie-olsen commented:
@jgrosspietsch updated with your refactoring of the function logic, added return type and also added an interface to assign to the function param. Everything seems to be working as expected on the front-end (+ unit tests passing) after local testing.

jgrosspietsch reviewed (approved):
Hell yeah. Thanks!

:cool:
